framework/src/modules/interoperability/utils.ts

```diff
@@ -106,3 +146,295 @@ export const computeValidatorsHash = (
 
 export const sortValidatorsByBLSKey = (validators: ActiveValidators[]) =>
 	validators.sort((a, b) => a.blsKey.compare(b.blsKey));
+
+export const rawStateStoreKey = (storePrefix: number) => {
+	const moduleIDBuffer = Buffer.alloc(4);
+	moduleIDBuffer.writeInt32BE(MODULE_ID_INTEROPERABILITY, 0);
+	const storePrefixBuffer = Buffer.alloc(2);
+	storePrefixBuffer.writeUInt16BE(storePrefix, 0);
+
+	return Buffer.concat([DB_KEY_STATE_STORE, moduleIDBuffer, storePrefixBuffer]);
+};
+
+export const checkLivenessRequirementFirstCCU = (
+	partnerChainAccount: ChainAccount,
+	txParams: CrossChainUpdateTransactionParams,
+): VerificationResult => {
+	if (partnerChainAccount.status === CHAIN_REGISTERED && txParams.certificate.equals(EMPTY_BYTES)) {
+		return {
+			status: VerifyStatus.FAIL,
+			error: new Error(
+				`Sending partner chain ${txParams.sendingChainID} is in registered status so certificate cannot be empty.`,
+			),
+		};
+	}
+
+	return {
+		status: VerifyStatus.OK,
+	};
+};
+
+export const checkCertificateValidity = (
+	partnerChainAccount: ChainAccount,
+	encodedCertificate: Buffer,
+): VerificationResult => {
+	if (!encodedCertificate.equals(EMPTY_BYTES)) {
+		const decodedCertificate = codec.decode<Certificate>(certificateSchema, encodedCertificate);
+		if (
+			decodedCertificate.blockID.equals(EMPTY_BYTES) ||
+			decodedCertificate.stateRoot.equals(EMPTY_BYTES) ||
+			decodedCertificate.validatorsHash.equals(EMPTY_BYTES) ||
+			decodedCertificate.timestamp === 0
+		) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Certificate is missing required values.'),
```

Comment:
[Optional] Maybe these function can return type `Error | undefined`

Reply:
I was doing that but then I saw it was looking more simplified in this form, also we need to capture different errors coming from these functions